examples/wave/wave-op-mpi.py

```diff
@@ -192,14 +196,19 @@ def main(ctx_factory, dim=2, order=3, visualize=False):
     def rhs(t, w):
         return wave_operator(dcoll, c=c, w=w)
 
+    compiled_rhs = actx.compile(rhs)
+
     if comm.rank == 0:
         logger.info("dt = %g", dt)
 
     t = 0
     t_final = 3
     istep = 0
     while t < t_final:
-        fields = rk4_step(fields, t, dt, rhs)
+        if lazy:
+            fields = thaw(freeze(fields, actx), actx)
```

Comment:
This is odd, and it shouldn't be here. What happens if you remove it?

Reply:
This is directly related to this discussion: https://github.com/inducer/grudge/pull/185#discussion_r757025159. Unresolved so we can continue the discussion there.